analisis de datos del archivo 

In [2]:
import pandas as pd

In [17]:
archivo="Datos/Cobertura_Vacunacion_Valle_Del_Cauca_2022-2023.xlsx"
df = pd.read_excel(archivo)
print(df.head())

    ANO    MES  CODIGO DEPARTAMENTO     DEPARTAMENTO  CODIGO MUNICIPIO  \
0  2022  ENERO                   76  VALLE DEL CAUCA             76020   
1  2022  ENERO                   76  VALLE DEL CAUCA             76020   
2  2022  ENERO                   76  VALLE DEL CAUCA             76020   
3  2022  ENERO                   76  VALLE DEL CAUCA             76020   
4  2022  ENERO                   76  VALLE DEL CAUCA             76020   

  MUNICIPIO GRUPO POBLACIONAL    BIOLOGICO  META  DOSIS APLICADAS  COBERTURA  \
0    ALCALÁ     RECIEN NACIDO          BCG   153                0   0.000000   
1    ALCALÁ     RECIEN NACIDO  HEPATITIS B   153                0   0.000000   
2    ALCALÁ         DOS MESES    POLIO 1ra   153               12   7.843137   
3    ALCALÁ      CUATRO MESES    POLIO 2da   153               14   9.150327   
4    ALCALÁ        SEIS MESES    POLIO 3ra   153               17  11.111111   

   DOSIS APLICADAS ACUMULADAS  COBERTURA ACUMULADA  
0                    

El archivo contiene información sobre la aplicación de biológicos en el municipio de Alcalá, departamento del Valle del Cauca, correspondiente al mes de enero del año 2022. Cada fila representa una combinación única de grupo poblacional y vacuna aplicada, con su respectiva meta de cobertura, dosis administradas y los porcentajes de cobertura alcanzados, tanto mensual como acumuladamente. En los dos primeros registros, correspondientes a recién nacidos y las vacunas BCG y Hepatitis B, no se reportan dosis aplicadas, lo que genera coberturas en cero. En los registros siguientes se observa la administración de vacunas correspondientes a los dos, cuatro y seis meses de edad, con coberturas que oscilan entre el 7.84% y el 11.11%. Esto indica un bajo nivel de cumplimiento respecto a la meta establecida 153.

Tipos de Variables

In [ ]:
# Mostrar tipos únicos de datos 
print(" Tipos de datos únicos en el DataFrame ")
print(df.dtypes.value_counts())

# Mostrar tipo de cada columna
print("\n Tipo de dato por columna ")
print(df.dtypes)

=== Tipos de datos únicos en el DataFrame ===
int64      6
object     5
float64    2
dtype: int64

=== Tipo de dato por columna ===
ANO                             int64
MES                            object
CODIGO DEPARTAMENTO             int64
DEPARTAMENTO                   object
CODIGO MUNICIPIO                int64
MUNICIPIO                      object
GRUPO POBLACIONAL              object
BIOLOGICO                      object
META                            int64
DOSIS APLICADAS                 int64
COBERTURA                     float64
DOSIS APLICADAS ACUMULADAS      int64
COBERTURA ACUMULADA           float64
dtype: object


El DataFrame contiene un total de 13 columnas distribuidas en tres tipos de datos: 6 columnas son de tipo entero (int64), 5 son de tipo texto (object) y 2 son de tipo decimal (float64). Las variables numéricas enteras corresponden a identificadores, cantidades aplicadas y metas, mientras que las variables tipo float64 representan porcentajes de cobertura. Las columnas categóricas como MES, DEPARTAMENTO, MUNICIPIO, GRUPO POBLACIONAL y BIOLÓGICO son de tipo object.

Unicidad

In [ ]:


print("Número total de filas:", len(df))
print("Filas duplicadas completamente:", df.duplicated().sum())

# Verificar unicidad por columnas clave
columnas_clave = ['ANO', 'MES', 'CODIGO DEPARTAMENTO', 'CODIGO MUNICIPIO', 'GRUPO POBLACIONAL', 'BIOLOGICO']
duplicados_clave = df.duplicated(subset=columnas_clave).sum()
print("Filas duplicadas por claves:", duplicados_clave)


Número total de filas: 24960
Filas duplicadas completamente: 0
Filas duplicadas por claves: 0


El archivo contiene 24.960 registros y cumple con el criterio de unicidad, ya que no presenta filas duplicadas ni repeticiones en la combinación clave de año, mes, municipio, grupo poblacional y biológico, lo que indica una correcta estructura de los datos.

Validez

In [9]:

# Revisar rangos y formatos esperados
df['COBERTURA'] = df['COBERTURA'].astype(str).str.replace(",", ".").astype(float)
df['COBERTURA ACUMULADA'] = df['COBERTURA ACUMULADA'].astype(str).str.replace(",", ".").astype(float)

print("Cobertura fuera de rango [0–100]:", df[(df['COBERTURA'] < 0) | (df['COBERTURA'] > 100)].shape[0])
print("Cobertura acumulada fuera de rango [0–100]:", df[(df['COBERTURA ACUMULADA'] < 0) | (df['COBERTURA ACUMULADA'] > 100)].shape[0])

# Meses válidos
meses_validos = ['ENERO', 'FEBRERO', 'MARZO', 'ABRIL', 'MAYO', 'JUNIO',
                 'JULIO', 'AGOSTO', 'SEPTIEMBRE', 'OCTUBRE', 'NOVIEMBRE', 'DICIEMBRE']
print("Valores inválidos en columna MES:", df[~df['MES'].isin(meses_validos)].shape[0])



Cobertura fuera de rango [0–100]: 1
Cobertura acumulada fuera de rango [0–100]: 511
Valores inválidos en columna MES: 0


Analisis 

presenta una alta validez general en sus datos: solo se identificó 1 valor fuera del rango esperado (0–100) en la columna de cobertura y 511 valores fuera de rango en la cobertura acumulada, lo cual debe revisarse. No se encontraron valores inválidos en la columna de mes, lo que indica consistencia con los nombres esperados.

CONSISTENCIA 

In [12]:

# Detectar posibles inconsistencias en variables tipo texto
print("\nRevisión de valores inconsistentes:")
for col in df.select_dtypes(include='object').columns:
    print(f"\n{col} - valores únicos:")
    print(df[col].value_counts(dropna=False))



Revisión de valores inconsistentes:

MES - valores únicos:
ENERO         2080
FEBRERO       2080
MARZO         2080
ABRIL         2080
MAYO          2080
JUNIO         2080
JULIO         2080
AGOSTO        2080
SEPTIEMBRE    2080
OCTUBRE       2080
NOVIEMBRE     2080
DICIEMBRE     2080
Name: MES, dtype: int64

DEPARTAMENTO - valores únicos:
VALLE DEL CAUCA    24960
Name: DEPARTAMENTO, dtype: int64

MUNICIPIO - valores únicos:
ALCALÁ                 624
ANDALUCÍA              624
LA VICTORIA            624
OBANDO                 624
PALMIRA                624
PRADERA                624
RESTREPO               624
RIOFRÍO                624
ROLDANILLO             624
SAN PEDRO              624
SEVILLA                624
TORO                   624
TRUJILLO               624
TULUÁ                  624
ULLOA                  624
VERSALLES              624
VIJES                  624
YOTOCO                 624
YUMBO                  624
LA UNIÓN               624
LA CUMBRE              624
JA

Hay un nivel adecuado de consistencia en las variables categóricas evaluadas. La columna MES presenta exactamente los doce meses del año, todos en mayúsculas y uniformemente distribuidos con 2080 registros cada uno, lo que indica una codificación homogénea. La variable DEPARTAMENTO contiene únicamente el valor “VALLE DEL CAUCA”, como es esperable dado el enfoque regional del archivo. En la columna MUNICIPIO, aunque hay múltiples valores, no se detectan errores de escritura ni inconsistencias aparentes en los nombres. De igual forma, la columna BIOLÓGICO muestra las diferentes vacunas y esquemas con formatos consistentes, como “POLIO 1ra” o “TRIPLE VIRAL ref 5 años”, sin duplicaciones tipográficas ni variaciones en la nomenclatura. Esto sugiere que el archivo ha sido correctamente estructurado y estandarizado en términos de sus variables categóricas.

COMPLETITUD 

In [11]:

# Revisar valores nulos y vacíos
print("\nValores nulos por columna:")
print(df.isnull().sum())

print("\nValores vacíos ('') por columna:")
for col in df.columns:
    vacios = df[col].astype(str).str.strip().eq('').sum()
    if vacios > 0:
        print(f"{col}: {vacios}")


Valores nulos por columna:
ANO                           0
MES                           0
CODIGO DEPARTAMENTO           0
DEPARTAMENTO                  0
CODIGO MUNICIPIO              0
MUNICIPIO                     0
GRUPO POBLACIONAL             0
BIOLOGICO                     0
META                          0
DOSIS APLICADAS               0
COBERTURA                     0
DOSIS APLICADAS ACUMULADAS    0
COBERTURA ACUMULADA           0
dtype: int64

Valores vacíos ('') por columna:


Este resultado indica que el conjunto de datos está completamente lleno, sin valores nulos ni celdas vacías en ninguna de sus columnas. Todas las variables, tanto categóricas como numéricas, presentan información completa en los 24.960 registros evaluados. 